libminifi/src/io/FileStream.cpp

```diff
@@ -101,13 +96,12 @@ int FileStream::writeData(std::vector<uint8_t> &buf, int buflen) {
 
 int FileStream::writeData(uint8_t *value, int size) {
   if (!IsNullOrEmpty(value)) {
-    std::lock_guard<std::recursive_mutex> lock(file_lock_);
+    std::lock_guard<std::mutex> lock(file_lock_);
     if (file_stream_->write(reinterpret_cast<const char*>(value), size)) {
       offset_ += size;
       if (offset_ > length_) {
         length_ = offset_;
       }
-      file_stream_->seekg(offset_);
```

Comment:
Hope I don't miss anything here due to lack of coffee, but it seems to be that there is no real point in seeking to where we just finished writing.

Reply:
This seeks the input position indicator after write has automatically seeked the output indicator. Yes, they are separate, and no, I don't find this intuitive either.
https://godbolt.org/z/oe1M4j

Reply:
I guess the `tellg` and `tellp` pointers could in theory be different, but they have never been, and if the standard library changes this, that will break almost all programs in existence, so I don't think they will. :)

Reply:
Yup, I assumed that `stringstream` would be similar to `fstream`, but apparently no. No problem in this case, thanks @fgerlits.